src/Cli/OrgCommands.php

```diff
@@ -132,29 +122,127 @@ public function orgAnalyze($org, $options = ['as' => 'default', 'format' => 'tab
     }
 
     /**
-     * Get support level from README.md contents.
+     * Guess codeowners content.
      */
-    protected static function getSupportLevel($readme_contents)
+    public function guessCodeOwners($api, $org, $repoName)
     {
-        $support_level = null;
-        $lines = explode("\n", $readme_contents);
-        $badges = SupportLevel::getSupportLevelBadges();
-        foreach ($lines as $line) {
-            foreach ($badges as $key => $badge) {
-                // Get the badge text from the badge markup.
-                preg_match('/^\[\!\[([A-Za-z\s\d]+)\]\(https:\/\/img.shields.io/', $badge, $matches);
-                if (!empty($matches[1])) {
-                    if (strpos($line, $matches[1]) !== false) {
-                        $support_level = $key;
-                        break 2;
+        $codeowners = [];
+        $ownerSource = '';
+        try {
+            $data = $api->gitHubAPI()->api('repo')->contents()->show($org, $repoName, 'CODEOWNERS');
+            if (!empty($data['content'])) {
+                $content = base64_decode($data['content']);
+                $ownerSource = 'file';
+                $codeowners = static::filterGlobalCodeOwners($content);
+            }
+        } catch (\Exception $e) {
+        }
+
+        return static::inferOwners($api, $org, $repoName, $codeowners, $ownerSource);
+    }
+
+    /**
+     * @command org:update-projects-info
+     * @param $csv_file The path to csv file that contains projects information.
+     */
+    public function orgUpdateProjectsInfo($csv_file, $options = [
+        'as' => 'default',
+        'update-codeowners' => false,
+        'update-support-level-badge' => false,
+        'branch-name' => 'project-update-info',
+        'commit-message' => 'Update project information.',
+        'pr-body' => '',
+    ])
+    {
+        $api = $this->api($options['as']);
+        $updateCodeowners = $options['update-codeowners'];
+        $updateSupportLevelBadge = $options['update-support-level-badge'];
+        if (!$updateCodeowners && !$updateSupportLevelBadge) {
+            throw new \Exception("Either --update-codeowners or --update-support-level-badge must be specified.");
+        }
+        $prTitle = '[UpdateTool - Project Information] Update project information.';
+        $branchName = $options['branch-name'];
+        $commitMessage = $options['commit-message'];
+        if (file_exists($csv_file)) {
+            $csv = new \SplFileObject($csv_file);
+            $csv->setFlags(\SplFileObject::READ_CSV);
+            foreach ($csv as $row_id => $row) {
+                // Skip header row.
```

Comment:
Instead of skipping the header row, it would be better to call a search function that returned `[3, 5, 23, 97]` based on the index that each header row label is found at.

Reply:
you mean using the header row to lookup for the right column number based on column titles?

Reply:
Yep. Then the script does not become fragile if the composition of the spreadsheet changes over time (as long as the titles of the columns we depend on remain constant)